server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -95,5 +95,5 @@ class AddressIndexRepository @Inject()(conf : AddressIndexConfigModule, elasticC
         )
       )
     }
-  }.map(_.as[PostcodeAddressFileAddress])
+  }.map(response => (response.as[PostcodeAddressFileAddress], response.maxScore))
 }
```

Comment:
@Mironor  this is what i mean, generally i only return tuples when i map that function, then construct a map after. even then i dont like doing it. Return types are preferred across the board